argo/providers/kfp/provider.go

```diff
@@ -91,6 +92,13 @@ func (kfpp KfpProvider) DeletePipeline(ctx context.Context, providerConfig KfpPr
 		Context: ctx,
 	}, nil)
 
+	if err != nil {
+		var errorResult *pipeline_service.DeletePipelineDefault
+		if errors.As(err, &errorResult) && errorResult.Payload.Code == kfpApiConstants.KfpResourceNotFoundCode {
+			err = nil
```

Comment:
I wouldn't re-assign the error and instead `return nil` (same below)

Reply:
I personally didnt like the multiple return points and to make that possible with golang need to use the mutability possibility...

Reply:
I don't necessarily like it either, but I prefer it over mutable errors. It is very [go-y](https://medium.com/swlh/return-early-pattern-3d18a41bba8) though 😓